crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -73,6 +73,10 @@ pub trait Struct: Reflect {
 
     /// Clones the struct into a [`DynamicStruct`].
     fn clone_dynamic(&self) -> DynamicStruct;
+
+    /// Returns the structs [`StructInfo`].
+    /// Will return `None` if [`TypeInfo`] is not available.
+    fn get_represented_kind_info(&self) -> Option<&StructInfo>;
```

Comment:
Hm, now I also wonder if we should maybe give these each kind-specific names? That might make it easier to find and reason about. 🤔

For example, this could be `get_represented_struct_info`.

Thoughts?

Reply:
Yeah, I think that's a bit less error prone.

Reply:
I'll defer to you but: https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

Any reason to not go for the short and simple `.struct_info()`?

Reply:
> I'll defer to you but: https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter
> 
> 
> 
> Any reason to not go for the short and simple `.struct_info()`?

I do prefer the shorter version, but for now I think we should stick with the longer version so that things are consistent.

In a followup, maybe we rename all the `get_represented_***` methods to their shorter versions?

Originally I had it to better differentiate with the method on `Typed` and to make it clear that this is the _represented_ info. But with proper documentation, I think can simplify (in a breaking-change followup PR).

Reply:
Understandable :+1: 

Pushed the rest of the impls (except `ValueInfo`, not sure if that fits into this) using the long form names for now.

I don't mind doing the renaming chore in a follow up later if it's desired to do it. Imo. the long names are a bit noisy.

Reply:
> directly implemented inherently on `dyn Struct`

Not sure I'm following- could you elaborate?